src/client/components/DiceTemplates.tsx

```diff
@@ -954,60 +978,112 @@ function ModifierNumberEditor({
+        onChange={(newValue: string) =>
+          dispatch({
+            actions: [
+              updatePartAction(myId, part.id, categoryId, template, {
+                proficiency: parseFloat(
+                  newValue
+                ) as typeof proficiencyValues[number],
+              }),
+            ],
+            optimisticKey: "proficiency",
+            syncToServerThrottle: DEFAULT_SYNC_TO_SERVER_DEBOUNCE_TIME,
+          })
+        }
+      />
+    </label>
   );
+}
 
-  if (!template) {
-    return null;
-  }
+function TemplateNoteEditor({
+  categoryId,
+  template,
+}: {
+  categoryId: RRDiceTemplateCategoryID;
+  template: RRDiceTemplate;
+}) {
+  const dispatch = useServerDispatch();
+  const myself = useMyProps("diceTemplateCategories", "id");
 
   return (
     <div>
-      {template.categoryIndex !== -1 && (
-        <label>
-          Category:
-          {diceCategories.map((category, categoryIndex) => (
-            <label key={categoryIndex} className="radio-label">
-              <input
-                type="radio"
-                name="categoryIndex"
-                value={categoryIndex}
-                checked={categoryIndex === template.categoryIndex}
-                onChange={(e) =>
-                  dispatch({
-                    actions: [
-                      diceTemplateUpdate({
-                        id: templateId,
-                        changes: {
-                          categoryIndex: clamp(
-                            0,
-                            categoryIndex,
-                            diceCategories.length - 1
-                          ),
-                        },
-                      }),
-                    ],
-                    optimisticKey: "categoryIndex",
-                    syncToServerThrottle: DEFAULT_SYNC_TO_SERVER_DEBOUNCE_TIME,
-                  })
-                }
-              />
-              <FontAwesomeIcon icon={category} fixedWidth />
-            </label>
-          ))}
-        </label>
-      )}
+      <label>
+        Category:
+        {myself.diceTemplateCategories.map((category) => (
+          <label
+            key={category.id}
+            className="radio-label"
+            title={category.categoryName}
+          >
+            <input
+              type="radio"
+              name="icon"
+              value={category.id}
+              //TODO how was this working?
+              checked={categoryId === category.id}
+              onChange={(e) =>
+                dispatch({
+                  actions: [
+                    //TODO does this work?
+                    playerRemoveDiceTemplate({
+                      id: myself.id,
+                      categoryId: category.id,
+                      templateId: template.id,
+                    }),
+                    playerAddDiceTemplate({
+                      id: myself.id,
+                      categoryId: category.id,
+                      template: template,
```

Comment:
Good question :D

Reply:
It should now. Might be a more elegant way of doing this though...